spec/unit/pops/types/p_sem_ver_type_spec.rb

```diff
@@ -0,0 +1,217 @@
+      end
+
+      it "can be matched against a versions in 'in' expression" do
+        code = <<-CODE
+          notice(SemVer('1.1.1') in [SemVer('1.0.0'), SemVer('1.1.1'), SemVer('2.3.4')])
+        CODE
+        expect(eval_and_collect_notices(code)).to eql(['true'])
+      end
+
+      it "can be matched against a VersionRange using an 'in' expression" do
+        code = <<-CODE
+          notice(SemVer('1.1.1') in SemVerRange('>1.0.0'))
+        CODE
+        expect(eval_and_collect_notices(code)).to eql(['true'])
+      end
+
+      it "can be matched against multiple VersionRanges using an 'in' expression" do
+        code = <<-CODE
+          notice(SemVer('1.1.1') in [SemVerRange('>=1.0.0 <1.0.2'), SemVerRange('>=1.1.0 <1.1.2')])
+        CODE
+        expect(eval_and_collect_notices(code)).to eql(['true'])
+      end
+    end
+
+    context 'matching SemVer' do
+      suitability = {
+        [ '1.2.3',         '1.2.2' ] => false,
+        [ '>=1.2.3',       '1.2.2' ] => false,
+        [ '<=1.2.3',       '1.2.2' ] => true,
+        [ '1.2.3 - 1.2.4', '1.2.2' ] => false,
+        [ '~1.2.3',        '1.2.2' ] => false,
+        [ '~1.2',          '1.2.2' ] => true,
+        [ '~1',            '1.2.2' ] => true,
+        [ '1.2.x',         '1.2.2' ] => true,
+        [ '1.x',           '1.2.2' ] => true,
+
+        [ '1.2.3',         '1.2.3-alpha' ] => true,
+        [ '>=1.2.3',       '1.2.3-alpha' ] => true,
+        [ '<=1.2.3',       '1.2.3-alpha' ] => true,
+        [ '>1.2.3',        '1.2.3-alpha' ] => false,
+        [ '<1.2.3',        '1.2.3-alpha' ] => false,
+        [ '1.2.3 - 1.2.4', '1.2.3-alpha' ] => true,
+        [ '1.2.3 - 1.2.4', '1.2.4-alpha' ] => true,
+        [ '1.2.3 - 1.2.4', '1.2.5-alpha' ] => false,
+        [ '~1.2.3',        '1.2.3-alpha' ] => true,
+        [ '~1.2.3',        '1.3.0-alpha' ] => false,
+        [ '~1.2',          '1.2.3-alpha' ] => true,
+        [ '~1.2',          '2.0.0-alpha' ] => false,
+        [ '~1',            '1.2.3-alpha' ] => true,
+        [ '~1',            '2.0.0-alpha' ] => false,
+        [ '1.2.x',         '1.2.3-alpha' ] => true,
+        [ '1.2.x',         '1.3.0-alpha' ] => false,
+        [ '1.x',           '1.2.3-alpha' ] => true,
+        [ '1.x',           '2.0.0-alpha' ] => false,
+
+        [ '1.2.3',         '1.2.3' ] => true,
+        [ '>=1.2.3',       '1.2.3' ] => true,
+        [ '<=1.2.3',       '1.2.3' ] => true,
+        [ '1.2.3 - 1.2.4', '1.2.3' ] => true,
+        [ '~1.2.3',        '1.2.3' ] => true,
+        [ '~1.2',          '1.2.3' ] => true,
+        [ '~1',            '1.2.3' ] => true,
+        [ '1.2.x',         '1.2.3' ] => true,
+        [ '1.x',           '1.2.3' ] => true,
+
+        [ '1.2.3',         '1.2.4' ] => false,
+        [ '>=1.2.3',       '1.2.4' ] => true,
+        [ '<=1.2.3',       '1.2.4' ] => false,
+        [ '1.2.3 - 1.2.4', '1.2.4' ] => true,
+#        [ '~1.2.3',        '1.2.4' ] => true, Awaits fix for PUP-6242
```

Comment:
Ideally this should be a separate pending test (but we have so many that just scroll by that people have stopped caring)